content/blog/2022/2022-01-28-getting-started-with-amazon-kinesis.md

```diff
@@ -0,0 +1,858 @@
+        PutRecordsRequest putRecordsRequest 
+						        = PutRecordsRequest
+						        .builder()
+						        .streamName(Constants.MY_DATA_STREAM)
+						        .records(putRecordsRequestEntryList)
+						        .build();
+        
+		PutRecordsResponse putRecordsResult = kinesisClient
+				.putRecords(putRecordsRequest);
+		
+        logger.info("Put records Result" + putRecordsResult);
+        kinesisClient.close();
+	}
+
+	private static KinesisClient getKinesisClient() {
+      ...
+      ...
+	}
+}
+
+```
+
+Here we are first creating the request object for the `putRecords()` method with the name of the data stream, partition key, and sequence number. Running this program gives the following output:
+
+```shell
+...ResultPutRecordsResponse(FailedRecordCount=0, 
+	Records=[
+	PutRecordsResultEntry(SequenceNumber=49626569155656830268862440193770802392642928158972051474, ShardId=shardId-000000000001), 
+
+	PutRecordsResultEntry(SequenceNumber=49626569155656830268862440193772011318462542788146757650, ShardId=shardId-000000000001), 
+
+	PutRecordsResultEntry(SequenceNumber=49626569155656830268862440193773220244282157417321463826, ShardId=shardId-000000000001), 
+
+	PutRecordsResultEntry(SequenceNumber=49626569155656830268862440193774429170101772046496170002, ShardId=shardId-000000000001), 
+
+	PutRecordsResultEntry(SequenceNumber=49626569155656830268862440193775638095921386675670876178, ShardId=shardId-000000000001)])
+```
+In this output, we can see the same `shardId` for all the records added to the stream. This is because we are using the same partition key for all our records, they have been put into the same shard with shardId : `shardId-000000000001`.
+
+As mentioned before, other than the AWS SDK, we can use the Kinesis Producer Library (KPL) or the Kinesis agent for adding data to a Kinesis Data Stream:
+
+1. **Kinesis Producer Library (KPL)**: KPL is a library written in C++ for adding data into a Kinesis data stream. It runs as a child process to the main user process. So in case the child process stops due to any when connecting or writing to a Kinesis Data Stream, the main process continues to run.
+
+2. **Amazon Kinesis Agent**: Kinesis Agent is a stand-alone application that we can run as an agent on Linux-based server environments such as web servers, log servers, and database servers. The agent continuously monitors a set of files and collects and sends new data to Kinesis Data Streams. Please refer to the official documentation for guidance on configuring Kinesis Agent in a Linux-based server environment.
+
+### Data Consumption - Reading Data from Kinesis Data Streams
+We can develop two types of consumers for Kinesis Data Streams: shared fan-out consumers which share the throughput of the stream and enhanced fan-out consumers with dedicated throughput. To learn about the differences between them, and to see how you can create each type of consumer, see Reading Data from Amazon Kinesis Data Streams.
```

Comment:
At then end of this paragraph, should that be a link? But it would link to the current section? Seems odd.